File: backend/app/services/mymemory.py
import httpx


async def translate(text: str, source: str, target: str) -> str:
    if not text.strip() or source == target:
        return text

    try:
        url = "https://api.mymemory.translated.net/get"
        params = {"q": text.strip(), "langpair": f"{source}|{target}"}
        async with httpx.AsyncClient(timeout=10) as client:
            resp = await client.get(url, params=params)
            if resp.is_success:
                data = resp.json()
                t = data.get("responseData", {}).get("translatedText", "")
                if t and "MYMEMORY WARNING" not in t.upper():
                    return t
    except Exception:
        pass

    return ""

File: backend/src/services/mymemory.js
export async function translate(text, source, target) {
  if (!text?.trim() || source === target) return text;

  const url = new URL("https://api.mymemory.translated.net/get");
  url.searchParams.set("q", text.trim());
  url.searchParams.set("langpair", `${source}|${target}`);

  try {
    const resp = await fetch(url, { signal: AbortSignal.timeout(10000) });
    if (!resp.ok) return "";

    const data = await resp.json();
    const t = data?.responseData?.translatedText;
    if (t && !t.toUpperCase().includes("MYMEMORY WARNING")) return t;
  } catch {
    // fall through
  }

  return "";
}
